connector/src/main/scala/com/microsoft/kusto/spark/utils/ContainerProvider.scala

```diff
@@ -57,17 +62,25 @@ class ContainerProvider(
       .build
   }
 
-  def getContainer: ContainerAndSas = {
+  def getContainer(maybeIngestionStorageParams: Option[Array[IngestionStorageParameters]] = None)
+      : ContainerAndSas = {
     // Refresh if storageExpiryMinutes have passed since last refresh for this cluster as SAS should be valid for at least 120 minutes
     val now = Instant.now(Clock.systemUTC())
     val secondsElapsed =
       now.getEpochSecond - lastRefresh.getEpochSecond // get the seconds between now and last refresh
-    if (storageUris.isEmpty ||
-      secondsElapsed > cacheExpirySeconds /* If the cache has elapsed , refresh */ ) {
-      refresh()
-    } else {
-      roundRobinIdx = (roundRobinIdx + 1) % storageUris.size
-      storageUris(roundRobinIdx)
+    maybeIngestionStorageParams match {
+      case Some(ingestionStorageParams) =>
+        processContainerResults(
+          mutable.Buffer(
+            ContainerProvider.refreshUserSas(ingestionStorageParams, cacheExpirySeconds)))
+      case None =>
+        if (storageUris.isEmpty ||
+          secondsElapsed > cacheExpirySeconds /* If the cache has elapsed , refresh */ ) {
+          refresh()
+        } else {
+          roundRobinIdx = (roundRobinIdx + 1) % storageUris.size
+          storageUris(roundRobinIdx)
+        }
```

Comment:
```suggestion
    if (storageUris.isEmpty ||
      secondsElapsed > cacheExpirySeconds /* If the cache has elapsed , refresh */ ) {
      maybeIngestionStorageParams match {
      case Some(ingestionStorageParams) =>
        processContainerResults(
          mutable.Buffer(
            ContainerProvider.refreshUserSas(ingestionStorageParams, cacheExpirySeconds)))
      case None =>
          refresh()
        }
    }
    else {
      roundRobinIdx = (roundRobinIdx + 1) % storageUris.size
      storageUris(roundRobinIdx)
    }
```

Reply:
actually this will now have the problem of using only 1 container at a time

Reply:
I tried to fix it in the ContainerProvider class. Do have a look and provide if this is a good option :)